### DISCUSSION TITLE
Casts on aggregate fields

### DISCUSSION THREAD

**Discussion by R00118189** [filamentphp/filament#5378] [Ideas]
Is there a way to add a `Cast` class to the aggregated [field] https://filamentphp.com/docs/2.x/tables/columns/relationships#aggregating-relationships) maybe like 
`TextColumn::make('transactions_sum_value')->sum('transactions', 'value')->withCasts(['transactions_sum_value' => MoneyCast::class])`
the value on the `transactions` table is cast to MoneyCast::class, but when aggregated, it is returned as a string; therefore, need for casting on the field. When used as a standalone table, I apply it to the query, but the filament admin resource `field` does not have this option. I tried adding aggregate and cast to the `public static function getEloquentQuery(): Builder  {}` in the resource, but I believe it would be nicer if it is cast in the field.

**Comment by danharrin** [comment]
Use formatStateUsing() instead of a cast?

**Comment by R00118189** [comment]
It does not work for me as the aggregate result is a string, not a Cast
`TextColumn::make('transactions_sum_value')->sum('transactions', 'value')->->formatStateUsing(fn (\Brick\Money\Money $state): string => $state->formatTo(app()->getLocale())),`
I am using brick\money to localise all currency and date values better. Also, I store values in DB as minor units/cents.

**Reply by danharrin** [reply]
`->formatStateUsing(fn ($state): string => \Brick\Money\Money::of($state, 'USD')->formatTo(app()->getLocale())`

**Reply by R00118189** [reply]
Thank you @danharrin - I changed it a bit to allow for `null` aggregates and grab record currency, so it is fully dynamic
`TextColumn::make('transactions_sum_value')->sum('transactions', 'value')->formatStateUsing(fn ($state, $record): string => \Brick\Money\Money::ofMinor((int)$state, $record->currency)->formatTo(app()->getLocale())),
`
